sklearn/manifold/_isomap.py

```diff
@@ -294,7 +294,14 @@ def _fit_transform(self, X):
 
         self.dist_matrix_ = shortest_path(nbg, method=self.path_method, directed=False)
 
+        if np.array(X).dtype == np.float32:
+            self.dist_matrix_ = self.dist_matrix_.astype(X.dtype, copy=False)
+
         G = self.dist_matrix_**2
+
+        if np.array(X).dtype == np.float32:
+            G = G.astype(X.dtype, copy=False)
```

Comment:
I think this isn't needed. With `G = self.dist_matrix_**2` and `self.dist_matrix_` already of type `float32`, I think `G` will also be `float32`.

Reply:
@betatim is right and this can be removed.

Reply:
```suggestion
```